storage/sas_token.go

```diff
@@ -42,6 +44,10 @@ func ComputeAccountSASToken(accountName string,
 	stringToSign += signedProtocol + "\n"
 	stringToSign += signedVersion + "\n"
 
+	if signedVersion >= "2020-12-06" {
```

Comment:
since this is a string this comparison isn't guaranteed to work? might be worth conditionally appending this field if the users specified it, and then calling this out in the documentation instead?

Reply:
If the version string is guaranteed to be consisten on the format, then the comparison should work as expected. The field can't be appended conditionally, e.g. for version equal or later than 2020-12-06, even for empty `signedEncryptionScope`, the new line is necessary to be appended, otherwise the SAS will be rejected by the service.

Reply:
@tombuildsstuff tested this in go playground, and @magodo is correct (go version = 1.19 and 1.20)


![image](https://user-images.githubusercontent.com/492603/216716513-63c048c8-fae6-41f9-92ad-8985b47db288.png)

Reply:
@magodo apologies for the super delayed re-review here - I'd missed this one hadn't been re-reviewed/merged - if we can add a unit test to check this one it should otherwise be good to merge